scala/private/rule_impls.bzl

```diff
@@ -739,6 +745,23 @@ def _scala_binary_common(
       rjars,  #calling rules need this for the classpath in the launcher
       runfiles = runfiles)
 
+def _pack_sources(ctx):
+  java_srcs = [
```

Comment:
@hmemcpy why do we need this? Can’t we just use ctx.files.srcs?

Reply:
I tried, `ctx.files.srcs` holds a `list` of files, and that, for some reason, wasn't acceptable for `pack_sources`. But this may have been before I made it work, I'll try again to see if I can use it!

Reply:
I tried, `ctx.files.srcs` holds a `list` of files, and that, for some reason, wasn't acceptable for `pack_sources`. But this may have been before I made it work, I'll try again to see if I can use it!

Reply:
I tried, `ctx.files.srcs` holds a `list` of files, and that, for some reason, wasn't acceptable for `pack_sources`. But this may have been before I made it work, I'll try again to see if I can use it!

Reply:
@sha1n any chance you can take a look at this comment?
Additionally do we need two source jars? We generate a scala bytecode jar and a java one. Do we need two source jars (for intellij I mean)?
If not then why do we do these two iterations. Sounds like `ctx.files.srcs` should work. Also according to https://docs.bazel.build/versions/master/skylark/lib/ctx.html#files and https://docs.bazel.build/versions/master/skylark/depsets.html

Reply:
Sure! I missed that..

Reply:
Yes, that seems to be working just fine. However, I'm trying to get libraries with mixed scala/java sources to work smoothly, so if I end up creating two separate source jars, I might keep the current approach.